R/derive_var_confirmation_flag.R

```diff
@@ -0,0 +1,350 @@
+#' Derive Confirmation Flag
+#'
+#' Derive a flag which depends on other observations of the dataset. For
+#' example, flagging events which need to be confirmed by a second event.
+#'
+#' An example usage might be flagging if a patient received two required
+#' medications within a certain timeframe of each other.
+#'
+#' In the oncology setting, for example, the function could be used to flag if a
+#' response value can be confirmed by an other assessment. This is commonly
+#' used in endpoints such as best overall response.
+#'
+#' @param dataset Input dataset
+#'
+#'   The variables specified by the `by_vars` and `join_vars` parameter are
+#'   expected.
+#'
+#' @param by_vars By variables
+#'
+#'   The specified variables are used as by variables for joining the input
+#'   dataset with itself.
+#'
+#' @param order Order
+#'
+#'   The observations are ordered by the specified order.
+#'
+#' @param new_var New variable
+#'
+#'   The specified variable is added to the input dataset.
+#'
+#' @param join_vars Variables to keep from joined dataset
+#'
+#'   The variables needed from the other observations should be specified
+#'   for this parameter. The specified variables are added to the joined dataset
+#'   with suffix ".join". For example to flag all observations with `AVALC ==
+#'   "Y"` and `AVALC == "Y"` for at least one subsequent visit `join_vars =
+#'   vars(AVALC, AVISITN)` and `filter = AVALC == "Y" & AVALC.join == "Y" &
+#'   AVISITN < AVISITN.join` could be specified.
+#'
+#'   The `*.join` variables are not included in the output dataset.
+#'
+#' @param join_type Observations to keep after joining
+#'
+#'   The argument determines which of the joined observations are kept with
+#'   respect to the original observation. For example, if `join_type =
+#'   "at_after"` is specified all observations at and after the original
+#'   observations are kept.
+#'
+#'   For example for confirmed response or BOR in the oncology setting or
+#'   confirmed deterioration in questionnaires the confirmatory assessment must
+#'   be after the assessment to be flagged. Thus `join_type = "after"` could be
+#'   used.
+#'
+#'   Whereas, sometimes you might allow for confirmatory observations to occur
+#'   prior to the observation to be flagged. For example, to flag AEs occurring
+#'   on or after seven days before a COVID AE. Thus `join_type = "all"` could be
+#'   used.
+#'
+#'   *Default:* `"after"`
+#'
+#'   *Permitted Values:* `"before"`, `"at_before"`, `"at_after"`, `"after"`,
+#'   `"all"`
```

Comment:
I don't think there should be a default value here. I'd rather have people concisely select the `join_type`.

Reply:
Fine with me. @millerg23 , @rossfarrugia , do you agree?

Reply:
yes agree